src/js/server/devpmt-ppt-dispatchers.js

```diff
@@ -0,0 +1,290 @@
+    path: ["/editprefs/:npset"],
+    defaultTemplate: "editprefset",
+    rules: {
+        contextToExpose: {
+            commonTerms: {
+                "transform": {
+                    type: "fluid.transforms.free",
+                    func: "fluid.getForComponent",
+                    args: ["{gpii.devpmt}", "commonTermsDataSource.current.schemas"]
+                }
+            },
+            allSolutions: {
+                "transform": {
+                    type: "fluid.transforms.free",
+                    func: "fluid.getForComponent",
+                    args: ["{devpmt}", "model.solutions"]
+                }
+            }
+        }
+    },
+    invokers: {
+        contextPromise: {
+            funcName: "gpii.devpmt.editPrefSetHandler.contextPromise",
+            args: ["{that}", "{gpii.devpmt}", "{arguments}.0"]
+        },
+        checkAuthorization: {
+            funcName: "gpii.devpmt.ppt.checkAuthorization",
+            args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // req, res, next
+        }
+    }
+});
+
+/**
+ * Adds the `gpii.devpmt.npset` for the request to the handlebars context.
+ */
+gpii.devpmt.editPrefSetHandler.contextPromise = function (that, devpmt, req) {
+    var promTogo = fluid.promise();
+    fluid.promise.map(devpmt.fullPrefSetDataSource.get({prefsSafeId: req.params.npset}), function (data) {
+        if (!data.prefsSafe) {
+            promTogo.reject({
+                isError: true,
+                message: "Couldn't find preferences safe."
+            });
+            return;
+        };
+        var npset = devpmt.ontologyHandler.rawPrefsToOntology(data.prefsSafe.preferences, "flat");
+        var prefset = gpii.devpmt.npset({
+            npsetName: req.params.npset,
+            flatPrefs: npset,
+            docs: ""
+        });
+        // This data is coming from a convenience endpoint that includes
+        // the keys, but when we go back to save the prefset they would be in their
+        // own documents, so we are removing the keys to a separate data
+        // field here to make life easier in the frontend.
+        var prefsSafe = data.prefsSafe;
+        var keys = data.keys;
+        promTogo.resolve({
+            npset: prefset,
+            prefsSafe: prefsSafe,
+            keys: keys
+        });
+    });
+    return promTogo;
+};
+
+// URLPATH /saveprefset/:npset
+fluid.registerNamespace("gpii.devpmt.savePrefsetHandler");
+
+fluid.defaults("gpii.devpmt.savePrefsetHandler", {
+    gradeNames: ["gpii.express.middleware"],
+    path: ["/saveprefset/:npset"],
+    invokers: {
+        middleware: {
+            funcName: "gpii.devpmt.savePrefsetHandler.handleRequest",
+            args: ["{that}", "{devpmt}", "{arguments}.0", "{arguments}.1" /*, "{arguments}.2" */]
+        }
+    }
+});
+
+gpii.devpmt.savePrefsetHandler.handleRequest = function (that, devpmt, req /* res , next */) {
+    if (gpii.devpmt.ppt.loginHandler.checkAuthorization) {
```

Comment:
Does this way of calling `gpii.devpmt.ppt.loginHandler.checkAuthorization` function work? Again, please decouple the dependency on other components whenever possible. One alternative is to define a generic `checkAuthorization` utility to be shared by multiple components.

Reply:
Ooof, thanks for catching that.  It clearly does not work in that manner.